### PR TITLE
Sort out MinimalScene's mouse events

### LINKED ISSUE #293: View control mouse interaction issues
## Environment
* OS Version: Ubuntu 18.04
* Source or binary build?
 both binary and source, ignition fortress pre2
  
## Description

I ran into a few gui view control interaction issues with the mouse.  I don't see these issues using the old GzScene3D plugin. 

I can separate them into different issues if needed.

1. Incorrect mouse anchor point. 
  * This doesn't happen all the time but it happens often enough that it's quite easily reproducible on my machine. I also see incorrect anchor points for panning the camera from time to time, e.g. camera jumps when panning after resizing the render window or unexpected large camera pan movement because anchor point used is closer than actual point.

gui_view_anchor

2. Context menu triggered on mouse release after a zoom action
  * Looks like the menu is triggered on all right mouse button releases. We should not open the context menu after a mouse right button drag.
 
gui_view_context

3. Scroll zoom moves camera in wrong direction
  * The first zoom with scroll button works as expected. However, if you try to zoom again in the opposite direction, the **first scroll increment** will be in the wrong direction (still in the previous direction) correct.

gui_view_zoom

### PR DESCRIPTION
**Pull request by chapulina**
# 🦟 Bug fix

Fixes #293

## Summary

I started working on the issues pointed out on #293 and ran into various other issues and inconsistencies about the way we're handling mouse events. Currently there are scrolls and drags being sent as left clicks, both mouse presses and releases broadcasted as clicks... ~~I'm still sorting it out, hopefully will be done in time for the release.~~

Summary of changes:

* Add 3 new events: drag, scroll and press
* Press with any button is used to pick the orbit target
* Drag information was previously being acquired from a combination of the previous event and a left click. Now it's passed through the drag event.
* I think scroll info was also coming from other events, but now it's its own event
* The `HandleMouseViewControl` was a confusing function that re-sent the left clicks, so I removed it

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

**Comment by ahcorde** [comment]
This PR fixes: 
 - Scroll zoom moves camera in wrong direction
 - Context menu triggered on mouse release after a zoom action

I'm trying to debug -> Incorrect mouse anchor point.

File: src/plugins/minimal_scene/MinimalScene.cc [hunk 1]
```diff
@@ -431,31 +436,65 @@ void IgnRenderer::BroadcastRightClick()
...
+  events::KeyReleaseOnScene keyRelease(this->dataPtr->keyEvent);
+  App()->sendEvent(App()->findChild<MainWindow *>(), &keyRelease);
+
+  this->dataPtr->keyEvent.SetType(common::KeyEvent::NO_EVENT);
 }
 
 /////////////////////////////////////////////////
 void IgnRenderer::BroadcastKeyPress()
 {
   if (this->dataPtr->keyEvent.Type() == common::KeyEvent::PRESS)
```
Comment (jennuine) [inline src/plugins/minimal_scene/MinimalScene.cc:491]: Should this be `!=`?
```suggestion
  if (this->dataPtr->keyEvent.Type() != common::KeyEvent::PRESS)
```

**Review by jennuine — APPROVED** [review]
This looks good to me, some minor comments/questions

File: src/plugins/minimal_scene/MinimalScene.cc (hunk 1 above)
Comment (jennuine) [reply src/plugins/minimal_scene/MinimalScene.cc:491]: since the next line is a `return`

Comment (chapulina) [reply src/plugins/minimal_scene/MinimalScene.cc:491]: oh yeah! nice catch

**Review by jennuine — APPROVED** [review]

**Pull request merged**